dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorSpirvVulkan.java

```diff
@@ -0,0 +1,262 @@
+package com.dat3m.dartagnan.program.processing.compilation;
+
+import com.dat3m.dartagnan.program.event.Event;
+import com.dat3m.dartagnan.program.event.EventFactory;
+import com.dat3m.dartagnan.program.event.Tag;
+import com.dat3m.dartagnan.program.event.arch.vulkan.VulkanCmpXchg;
+import com.dat3m.dartagnan.program.event.arch.vulkan.VulkanRMW;
+import com.dat3m.dartagnan.program.event.arch.vulkan.VulkanRMWExtremum;
+import com.dat3m.dartagnan.program.event.arch.vulkan.VulkanRMWOp;
+import com.dat3m.dartagnan.program.event.core.ControlBarrier;
+import com.dat3m.dartagnan.program.event.core.GenericVisibleEvent;
+import com.dat3m.dartagnan.program.event.core.Load;
+import com.dat3m.dartagnan.program.event.core.Store;
+import com.dat3m.dartagnan.program.event.lang.spirv.*;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.dat3m.dartagnan.program.event.EventFactory.*;
+
+public class VisitorSpirvVulkan extends VisitorVulkan {
+
+    @Override
+    public List<Event> visitLoad(Load e) {
+        Event load = EventFactory.newLoad(e.getResultRegister(), e.getAddress());
```

Comment:
I think you could use `e.getCopy()` here and at many other places (e.g., for stores, fence events, control barriers, ...) to make the intention more clear and also reduce error sources.

Reply:
Well, I tried to be consistent with the other visitors